src/main/java/org/cactoos/iterable/SyncIterable.java

```diff
@@ -58,7 +64,16 @@ public SyncIterable(final X... src) {
      * @param iterable The iterable
      */
     public SyncIterable(final Iterable<X> iterable) {
-        super(new SyncScalar<>(() -> iterable));
+        this.iterable = new UncheckedScalar<>(
+            () -> iterable
+        );
+    }
+
+    @Override
+    public Iterator<X> iterator() {
+        synchronized (this.iterable) {
```

Comment:
@svendiedrichsen Why are you using the same object as a lock? Please, follow the same approach used by Yegor in `SyncInput` (a attribute to lock, and a second ctor).